distributions/demo-server/build.gradle

```diff
@@ -96,9 +108,84 @@ shadowJar {
 
 shadowJar.dependsOn(createSecretsFile)
 
-def copyResources() {
-    copy {
-        from 'resources/'
-        into 'src/main/resources'
+task createApiServerDockerfile(type: Dockerfile) {
+    description = 'Builds the API Server Dockerfile'
+    group = 'docker'
+    destFile project.file("${buildDir}/apiserver/Dockerfile")
+    from "${javaDockerContainer}"
+    exposePort 8080
+    copyFile("build/libs/demo-server-all.jar", "/app/demo-server-all.jar")
+    defaultCommand("java", "-jar", "/app/demo-server-all.jar")
+
+    // Read in secrets and set them as environment variables.
+    // NB: This is for demo purposes only. Production systems should manage and set secrets in a secure way.
+    def secretsFile = file(Paths.get(System.getProperty('user.home')).resolve('.gradle/dataportability.secrets.properties'))
```

Comment:
Should we create a constant at the beginning of the script for this path and document it for developers, e.g. dataportability.secrets.properties

Reply:
Yeah this was one of the things I wanted to discuss further. Originally for the local cloud extension we had secretes sourced from the main settings file in the .gradle home. I think it would be better if we actually separate all of the config related to dataportability out of that and into a sidefile that sits in the .gradle home directory along side the main settings file. That way, people can share their dataportability settings (if appropriate) without having to mix in ohter things they may have in the settings file.

@rtannenbaum This can be where ENV settings are stored that get baked into Dockerfiles for demo and other non-production (i.e. "non-sensitive") deployments.

Reply:
Nice. I agree and was thinking the same thing re checking in local settings rather than in the .gradle home where they can't be shared. (Or at least a template of it if we don't want to share local app secrets)

Reply:
I just added a note to Slack on a proposed approach to handle other config like URLs so we don't need to check them.